sdk/cosmosdb/cosmos/src/request/defaultAgent.ts

```diff
@@ -15,10 +15,22 @@ export let defaultHttpsAgent: Agent;
 if (isNode) {
   // tslint:disable-next-line:no-var-requires
   const https = require("https");
-  defaultHttpsAgent = new https.Agent({
-    keepAlive: true,
-    secureProtocol: "TLSv1_2_method"
-  });
+  // tslint:disable-next-line:no-var-requires
+  const tls = require("tls");
+
+  // minVersion only available in Node 10+
+  if (tls.DEFAULT_MIN_VERSION) {
+    defaultHttpsAgent = new https.Agent({
+      keepAlive: true,
+      minVersion: "TLSv1.2"
```

Comment:
should this be `tls.DEFAULT_MIN_VERISON` rather than hard-coded TLSv1.2?

Reply:
On Node 10, `tls.DEFAULT_MIN_VERISON` === "TLSv1"

Reply:
I guess I'm still not sold on hard-coding this value. It would be nice if the default path would use the highest available TLS version wouldn't it? TLSv1.3 is a thing...

Reply:
It still will use TLS 1.3 if available, that's part of the TLS negotiation. This just sets the min version that will be used.

Reply:
Yes of course. Carry on then.